src/Caching/StackExchangeRedis/src/RedisCache.cs

```diff
@@ -107,7 +108,7 @@ public void Set(string key, byte[] value, DistributedCacheEntryOptions options)
 
             var absoluteExpiration = GetAbsoluteExpiration(creationTime, options);
 
-            var result = _cache.ScriptEvaluate(SetScript, new RedisKey[] { _instance + key },
+            var result = _cache!.ScriptEvaluate(SetScript, new RedisKey[] { _instance + key },
```

Comment:
You can annotate `Connect` and `ConnectAsync` with `[MemberNotNull(nameof(_cache))]` to avoid the null forgiveness

Reply:
I had trouble applying MemberNotNull on ConnectAsync, we can chat about it when you are back.